js-classes/inheritance-es5.js

```diff
@@ -0,0 +1,171 @@
+const currentDate = new Date();
+
+const currentYear = currentDate.getFullYear();
+
+function Car(name, model, year, color, maxSpeed, fuelCapacity = 60, fuelConsumption = 10){
+	this.name = name;
+	this.model = model;
+	this.year = year;
+	this.color = color;
+	this.maxSpeed = maxSpeed;
+	this.fuelCapacity = fuelCapacity;
+	this.fuelConsumption = fuelConsumption;
+}
+
+Car.prototype.getFullName = function(){
+	return `Your car: ${this.name} ${this.model}`;
+}
+
+Car.prototype.getAge = function(){
+	return `Your car age: ${currentYear - this.year}`;
+}
+
+Car.prototype.changeColor = function(color){
+	if ((typeof(color) === 'string') && (isNaN(color))) {
+		if (color.toLowerCase() === this.color.toLowerCase()){
+			return 'The car is already painted in this color';
+		}
+
+		this.color = color.toLowerCase();
+		return `Color has been changed to ${this.color}`;
+	}
+
+	return 'Incorrect color value';
+}
+
+Car.prototype.calculateWay = function(kilometers, fuel){
+	if ((fuel > this.fuelCapacity) || (fuel < 0) || (typeof(fuel) !== 'number')) {
+		return 'Incorrect value fuel';
+	}
+
+	if ((kilometers < 0) || (typeof(kilometers) !== 'number'))  {
+		return 'Incorrect value kilometers';
+	}
+
+	if (fuel < 10){
+		console.log('Low level fuel, please refuel');
+	}
+
+	if ((fuel / kilometers) < (this.fuelConsumption / 100))	{
+		const neededFuel = kilometers * (this.fuelConsumption / 100);
+		let i = 0;
+		while (neededFuel > fuel){
+			i++;
+			fuel += this.fuelCapacity;
+		}
+		console.log(`Not enough fuel to ride, need to refuel ${i} times`);
+	}
+
+	const travelTime = ((kilometers / this.maxSpeed) * 60).toFixed(3);
+
+	const timeInHours = Math.floor(travelTime / 60);
+
+	const timeInMinutes = Math.round(travelTime - timeInHours * 60);
+
+	return `Travel time: ${timeInHours} hours and ${timeInMinutes} minutes`;
+}
+
+function extend(Child, Parent) {
+    Child.prototype = Object.create(Parent.prototype);
+    Child.prototype.constructor = Child;
+}
+
+function Nissan(name, model, year, color, maxSpeed, fuelCapacity = 60, fuelConsumption = 10, interiorColor){
+	Car.apply(this, arguments);
+	this.interiorColor = interiorColor;
+}
+
+extend(Nissan, Car);
+
+Nissan.prototype.changeInteriorColor = function(color){
```

Comment:
You don't need to create this method because your parent already has it. So you don't need `this.interiroColor` as well. To change color for the Nissan just use 
```
const nissan = new Nissan()

nissan.changeColor('red')

```

Reply:
The function of the parent changes the color of the car. The function specifically at nissan changes the color of its interior